### PR TITLE
Implement REST API based on ZeroEQ [DISCL-355]

### PR DESCRIPTION
**Pull request by rdumusc**

### CODE REVIEW

File: tide/master/rest/RestCommand.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+public:
+    /**
+     * Construct a REST command which can be exposed by a ZeroEQ http server.
+     *
+     * @param name The name of the command exposed through the REST interface,
+     *        separated by '::' instead of '/' (e.g. "restapi::mycommand").
+     */
+    explicit RestCommand( const std::string& name );
+
+    /** @retrun the name of the command passed in the constructor. */
```
Comment (jafyvilla) [inline tide/master/rest/RestCommand.h]: typo retrun

File: tide/master/rest/RestCommand.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    std::string getTypeName() const final;
+
+signals:
+    /** Emitted when the command has been recieved. */
```
Comment (jafyvilla) [inline tide/master/rest/RestCommand.h]: typo recieved

File: tide/master/rest/RestCommand.h (hunk 1 above)
Comment (rdumusc) [inline tide/master/rest/RestCommand.h]: done

File: tide/master/rest/RestCommand.h (hunk 2 above)
Comment (rdumusc) [inline tide/master/rest/RestCommand.h]: done

**Comment by jafyvilla** [comment]
+1 when the build gets green

**Comment by rdumusc** [comment]
Thanks, build was red due to old version of Servus (without Serializable) pulled in by Deflect. I swapped ZeroEQ and Deflect in .gitsubprojects as a fix.

File: CMakeLists.txt [hunk 3]
```diff
@@ -41,6 +41,11 @@ common_find_package(Qt5WebKitWidgets SYSTEM)
 common_find_package(Qt5Widgets REQUIRED)
 common_find_package(Qt5XmlPatterns REQUIRED)
 common_find_package(Qt5Xml REQUIRED)
+common_find_package(ZeroEQ)
+
+if(ZeroEQ_FOUND)
```
Comment (eile) [inline CMakeLists.txt]: if(TARGET ZeroEq)

File: tests/cpp/CMakeLists.txt [hunk 4]
```diff
@@ -16,10 +16,16 @@ set(TEST_LIBRARIES
   Qt5::OpenGL
 )
 
+if(NOT TIDE_ENABLE_REST_INTERFACE OR TRUE)
```
Comment (eile) [inline tests/cpp/CMakeLists.txt]: OR TRUE?

Comment (rdumusc) [inline tests/cpp/CMakeLists.txt]: ouch!

File: CMakeLists.txt (hunk 3 above)
Comment (rdumusc) [inline CMakeLists.txt]: If you prefer but it makes no difference :-)

Comment (eile) [inline CMakeLists.txt]: I have no preference, but it's my understanding that TARGETs are the cmake 3 way, and Find/FOUND is not advised.

**Pull request merged**